Added enum class and large scaled class

Isfirs opened the pull request:
Closes #1 
Closes #2
Closes #3 
Closes #5

Commit:
Added enum class and large scaled class

#1 #2

Commit:
Added long line constructor arguments

Commit:
Fixed arguments

Commit:
Changed fiel types

Commit:
Added imports

Commit:
Missing semicolon

Isfirs committed:
Fixed braces

Isfirs committed:
Added class docs and assigned variable

Isfirs committed:
Reached 180 characters

Isfirs committed:
Added space between field and class signature

cltnschlosser commented on src/org/company/LargeClass.java:
```diff
@@ -0,0 +1,25 @@
+ */
+public class LargeClass
+{
+    
```
This line is not needed. and then for everything below, there should only be one space between things for the long lines. EX:

``` java
private final String fieldOne   = "field one value";
private       String fieldTwo   = "field two value";
private final Object fieldThree = new Object();
```

cltnschlosser commented on src/org/company/MyClass.java:
```diff
@@ -1,5 +1,7 @@
 package org.company;
 
+import com.sun.istack.internal.NotNull;
```
What are these?

Isfirs replied:
I don't know. A method below is using them and they are were not imported. I used IntelliJ to auto-import them to resolve compiler problems.

cltnschlosser commented on src/org/company/MySuperLongClass.java:
```diff
@@ -0,0 +1,18 @@
+        //
+    }
+
+    public MySuperLongClass(final int one, final String two, final Object three, final long four, final double five, final float six, final MyEnum seven, final MyClass eight, final AbstractMyClass nine) {
```
This need to be split up. Its my preference that if we are going to split it up, that every parameter gets its own line.

Isfirs replied:
I will convert it. :)

Isfirs replied:
May  we group them in side the context when we hit a maximum line length?

like:

```
function(
var one,   var two
var three, var four
)
```

The lines are grouped in equal manners and aligned into a table grid.

cltnschlosser commented on src/org/company/LargeClass.java:
```diff
@@ -0,0 +1,25 @@
+ * @since Current.Project.Version
+ */
+public class LargeClass
+{
```
braces aren't consistent. We need to pick a style, I'm in favor of leaving this repo and braces on a new line, but am okay either way.

Isfirs replied:
Fixed already

cltnschlosser commented on src/org/company/MyEnum.java:
```diff
@@ -0,0 +1,11 @@
+    ENUM_ONE,
+    ENUM_TWO,
+    ENUM_THREE,
+    ENUM_FOUR,
```
Last enum value should end with semi colon.

Isfirs replied:
I agree. I will add 2 constructors, a default and a value plus field.

Isfirs committed:
Last enum should end with semicolon.

Isfirs committed:
Added a field and 2 constructors

- One constructor is using the field as parameter
- Assign parameter to field
- no 'private' for enum constructors, enum-constructors can't be anything else than private.

marvin-bitterlich committed:
Added formatter and formatted according to it

marvin-bitterlich committed:
fixed formatter and added stream examples
